### PR TITLE
mpd service: Start when needed and harden

### PR DESCRIPTION
**Pull request by jfrankenau**
###### Motivation for this change

By using a systemd socket it is possible to start MPD only when it's needed similar to the OpenSSH daemon config option `startWhenNeeded`.

Additionally, upstream provides a hardened systemd service file which I've integrated here. This also fixes two error messages from MPD regarding real-time priority and Avahi.

###### Things done

- [x] Tested using sandboxing
- [x]  Built on NixOS
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by 0xABAB — APPROVED** [review]

File: nixos/modules/services/audio/mpd.nix [hunk 1]
```diff
@@ -123,10 +133,24 @@ in {
 
   config = mkIf cfg.enable {
 
+    systemd.sockets.mpd = mkIf cfg.startWhenNeeded {
+      description = "Music Player Daemon Socket";
+      wantedBy = [ "sockets.target" ];
+      listenStreams = [
+        "/run/mpd/socket"
+        "${cfg.network.listenAddress}:${toString cfg.network.port}"
+      ];
```
Comment (Mic92) [inline nixos/modules/services/audio/mpd.nix]: Just to be sure: Did you test that mpd pick up multiple sockets, when passed via socket activation?

Comment (jfrankenau) [inline nixos/modules/services/audio/mpd.nix]: I have now and the upstream MPD socket file does the same. I decided to use `/var/run/mpd/socket` to use the same location MPD uses.

While testing I noticed that `any` for `listenAddress` needed to be handled with care which is fixed now.

Comment (0xABAB) [inline nixos/modules/services/audio/mpd.nix]: This is more of a general problem, but hardcoding "/run/mpd/socket" here and in other places is a not a modular solution. 

It means that if I later want to add another service which is also called `mpd`, but does something completely different that I need to care about the existence of this particular service; something a well designed system would not require.

Comment (jfrankenau) [inline nixos/modules/services/audio/mpd.nix]: You are right. I have removed the IPC socket for now. I couldn't find a client where the default is set to a local socket anyway. In the future we could add an option which allows to specify a path for it but I guess most people never even thought about using one.

**Pull request merged**